### PR TITLE
Playback 2023: completion rate story

### PR DESCRIPTION
**Pull request by leandroalonso**
| 📘 Part of: #1142 | 🎨 Designs: [Figma](https://www.figma.com/file/lH66LwxxgG8btQ8NrM0ldx/End-of-Year?type=design&node-id=1599-20385&mode=design) |
|:---:|:---:|

Adds the 2023 visual for the Completion Rate Story.

## Different states

1. Go to `CompletionRateStory.swift`
2. Enable Preview
3. ✅ Go over all the different percentages
4. You can also play with the numbers to see if the circle updates accordingly

## Story appearing

1. Make sure you're logged in to an account that has a few episodes listened
2. Go to Profile > Settings > Beta Features > enable `endOfYear`
3. Go back to Profile
4. Tap the EOY image
5. Skip to the penultimate story
5. ✅ Check that the story looks good compared to the design
6. Tap "Share this story" > Save Image
8. Go to Photos
9. ✅ The story image asset should be the last one in Photos
10. Give yourself Plus (Settings > Developer > Set to Paid Plus)
11. ✅ Check the story again, it should show a Plus badge
10. Give yourself Patron (Settings > Developer > Set to Paid Patron)
11. ✅ Check the story again, it should show a Patron badge

Please test on a big device (iPhone 15 Pro Max, for example), a normal device (iPhone 15), and a smaller device (iPod touch).

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

File: Modules/DataModel/Sources/PocketCastsDataModel/Private/Managers/EndOfYearDataManager.swift [hunk 1]
```diff
@@ -407,6 +434,20 @@ public struct YearOverYearListeningTime {
...
+    public let completed: Int
+
+    public var percentage: Double {
+        Double(completed) / Double(started)
+    }
+
+    public init(started: Int, completed: Int) {
+        self.started = started
+        self.completed = completed
+    }
```
Comment (emilylaguna) [inline Modules/DataModel/Sources/PocketCastsDataModel/Private/Managers/EndOfYearDataManager.swift]: Wdyt about:
- storing the computer the percent in init
- Clamping the percent to between 0 and 1
- If, for some reason the completed number if higher than the started, then use the completed number as the started. 

```suggestion
    public let percentage: Double
    
    public init(started: Int, completed: Int) {
        self.started = max(started, completed)
        self.completed = completed
        self.percentage = (Double(completed) / Double(started)).clamped(to: 0..<1)
    }
```

File: podcasts/End of Year/Stories/CompletionRateStory.swift [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+                               .scaleEffect(.init(width: -1, height: 1))
+                               .frame(width: geometry.size.height * 0.35, height: geometry.size.height * 0.35)
+                        )
+
+                        VStack(spacing: 0) {
+                            Text(percentageToDisplay)
+                            .font(.custom("DM Sans", size: geometry.size.height * 0.127).weight(.light))
+                            .foregroundColor(Color(red: 0.98, green: 0.98, blue: 0.99))
+
+                            Text("completion rate")
```
Comment (emilylaguna) [inline podcasts/End of Year/Stories/CompletionRateStory.swift]: L10n?

**Review by emilylaguna — CHANGES_REQUESTED** [review]
Nice. Left some comments below

**Comment by pocketcasts** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
EndOfYearDataManager.swift | 441 | Lines should not have trailing whitespace. (trailing_whitespace)


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/End of Year/Stories/CompletionRateStory.swift (hunk 2 above)
Comment (leandroalonso) [reply podcasts/End of Year/Stories/CompletionRateStory.swift]: Ops, done in 7d935120a099888648ad8bf55ffa24f34c910fe9

**Review by emilylaguna — APPROVED** [review]

**Pull request merged**